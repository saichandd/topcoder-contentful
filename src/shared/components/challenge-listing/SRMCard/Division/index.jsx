import PT from 'prop-types';
import React from 'react';
import LeaderboardAvatar from 'components/challenge-listing/LeaderboardAvatar';

import Tooltip from 'components/Tooltip';
import RegistrantsIcon from '../../Icons/RegistrantsIcon';
import SubmissionsIcon from '../../Icons/SubmissionsIcon';
import './style.scss';

function numRegistrantsTipText(number) {
  switch (number) {
    case 0: return 'No registrants';
    case 1: return '1 total registrant';
    default: return `${number} total registrants`;
  }
}

function numSubmissionsTipText(number) {
  switch (number) {
    case 0: return 'No submissions';
    case 1: return '1 total submission';
    default: return `${number} total submissions`;
  }
}
// Mock WINNERS array
let MOCK_WINNERS = [
  {
    handle: 'ACRush',
    photoURL: 'https://acrobatusers.com/assets/images/template/author_generic.jpg',
    isSmr: true,
  },
  {
    handle: 'tourist',
    photoURL: 'https://acrobatusers.com/assets/images/template/author_generic.jpg',
    isSmr: true,
  },
  {
    handle: 'RiaDWaW',
    isSmr: true,
  },
  {
    handle: 'KalininN',
    isSmr: true,
  },
  {
    handle: 'Lorem',
    photoURL: 'https://acrobatusers.com/assets/images/template/author_generic.jpg',
    isSmr: true,
  },
  {
    handle: 'Ipsum',
    isSmr: true,
  },
];

const MAX_VISIBLE_WINNERS = 5;

MOCK_WINNERS = MOCK_WINNERS.slice(0, MAX_VISIBLE_WINNERS);

const renderLeaderboard = MOCK_WINNERS.map(winner => (
  <div className="avatar-container" key={winner.handle}>
    <LeaderboardAvatar member={winner} />
    <div className="name">
      {winner.handle}
    </div>
  </div>
));

// mock resultsRows
const MOCK_RESULTS = [{
  title: 'The Basics Of Buying A Telescope',
  submission: 99,
  percent: 34.33,
}, {
  title: 'Shooting Stars',
  submission: 12,
  percent: 50,
}, {
  title: 'The Amazing Hubble',
  submission: 4,
  percent: 32.1,
}];

const resultsRows = MOCK_RESULTS.map(row => (
  <div className="results-detail-row" key={row.title}>
    <p className="d-title">
      {row.title}
    </p>
    <span className="challenge-stats">
      <span>
        <Tooltip content={numSubmissionsTipText(row.submission)}>
          <a className="num-sub" href="">
            <SubmissionsIcon />
            {' '}
            <span className="number">
              {row.submission}
            </span>
          </a>
        </Tooltip>
      </span>
      <span>
        <Tooltip content={numRegistrantsTipText(row.percent)}>
          <a className="num-reg" href="">
            <RegistrantsIcon />
            {' '}
            <span className="number">
              {row.percent}
%
            </span>
          </a>
        </Tooltip>
      </span>
    </span>
  </div>
));

/*
* Past SRMCard
*/
function Division({ division }) {
  return (
    <div className="division">
      <div className="leaderboard-row">
        <p className="stage">
          { division }
        </p>
        <div className="leaders">
          { renderLeaderboard}
        </div>
      </div>
      {resultsRows}
    </div>
  );
}

Division.propTypes = {
  division: PT.string.isRequired,
};

export default Division;
