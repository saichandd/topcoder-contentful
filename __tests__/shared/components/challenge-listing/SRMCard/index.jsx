import _ from 'lodash';
import mockStore from 'redux-mock-store';
import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import SRMCard from 'components/challenge-listing/SRMCard';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';

const store = mockStore()();

const mockDatas = [
  {
    category: 'now',
    srmChallenge: {},
  }, {
    category: 'upcoming',
    srmChallenge: {},
  }, {
    category: 'past',
    srmChallenge: {},
  },
];

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return (
      <StaticRouter context={{}}>
        <SRMCard {...this.props} />
      </StaticRouter>
    );
  }
}

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  _.forEach(mockDatas, (data) => {
    renderer.render((
      <Provider store={store}>
        <StaticRouter context={{}}>
          <SRMCard {...data} />
        </StaticRouter>
      </Provider>
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
    TU.renderIntoDocument((<Wrapper {...data} />));
  });
});
