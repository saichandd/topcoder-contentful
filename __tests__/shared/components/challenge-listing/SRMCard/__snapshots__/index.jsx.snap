// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <SRMCard
    category="now"
    srmChallenge={Object {}}
  />
</StaticRouter>
`;

exports[`Matches shallow shapshot 2`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <SRMCard
    category="upcoming"
    srmChallenge={Object {}}
  />
</StaticRouter>
`;

exports[`Matches shallow shapshot 3`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <SRMCard
    category="past"
    srmChallenge={Object {}}
  />
</StaticRouter>
`;
